test/lockup/lockup-s-token.ts

```diff
@@ -0,0 +1,1447 @@
+			dev.generatePropertyGroup(),
+			dev.generatePolicyFactory(),
+			dev.generatePolicyGroup(),
+		])
+		await dev.dev.mint(deployer, deployerBalance)
+		const policyAddress = await dev.generatePolicy('PolicyTestBase')
+		// eslint-disable-next-line @typescript-eslint/await-thenable
+		const policy = await artifacts.require('PolicyTestBase').at(policyAddress)
+		const propertyAddress = getPropertyAddress(
+			await dev.propertyFactory.create('test', 'TEST', user2, {
+				from: user2,
+			})
+		)
+		const [property] = await Promise.all([
+			artifacts.require('Property').at(propertyAddress),
+		])
+
+		await dev.addressConfig.setMetricsFactory(deployer)
+		await dev.metricsGroup.addGroup(
+			(
+				await dev.createMetrics(deployer, property.address)
+			).address
+		)
+
+		await dev.lockup.update()
+
+		return [dev, property]
+	}
+
+	const init2 = async (): Promise<
+		[DevProtocolInstance, PropertyInstance, number]
+	> => {
+		const [dev, property] = await init()
+		await dev.dev.approve(dev.lockup.address, 500)
+		await dev.lockup.depositToProperty(property.address, 100)
+		const tokenId = await dev.sTokenManager.latestTokenId()
+
+		return [dev, property, Number(tokenId)]
+	}
+
+	const err = (error: Error): Error => error
+
+	describe('Lockup; sTokensManager', () => {
+		it('get s tokens manager address', async () => {
+			const [dev] = await init()
+
+			const address = await dev.lockup.sTokensManager()
+			expect(address).to.be.equal(dev.sTokenManager.address)
+		})
+	})
+	describe('Lockup; depositToProperty', () => {
+		describe('success', () => {
+			it('get nft token.', async () => {
+				const [dev, property] = await init()
+				await dev.dev.approve(dev.lockup.address, 100)
+				await dev.lockup.depositToProperty(property.address, 100)
+				const owner = await dev.sTokenManager.ownerOf(1)
+				expect(owner).to.be.equal(deployer)
+				const position = await dev.sTokenManager.positions(1)
+				expect(position[0]).to.be.equal(property.address)
+				expect(position[1].toNumber()).to.be.equal(100)
+				// TODO ここ0でええんやろうか
+				expect(position[2].toNumber()).to.be.equal(0)
+				expect(position[3].toNumber()).to.be.equal(0)
+				expect(position[4].toNumber()).to.be.equal(0)
+			})
+			it('get 2 nft token.', async () => {
+				const [dev, property] = await init()
+				await dev.dev.approve(dev.lockup.address, 100)
+				await dev.lockup.depositToProperty(property.address, 100)
+				await dev.dev.approve(dev.lockup.address, 200)
+				await dev.lockup.depositToProperty(property.address, 200)
+				const owner = await dev.sTokenManager.ownerOf(2)
+				expect(owner).to.be.equal(deployer)
+				const position_1 = await dev.sTokenManager.positions(1)
+				const position = await dev.sTokenManager.positions(2)
+				expect(position[0]).to.be.equal(property.address)
+				expect(position[1].toNumber()).to.be.equal(200)
+				// TODO ここ0でええんやろうか
+				expect(position[2].toString()).to.be.equal(
+					'200000000000000000000000000000000000'
```

Comment:
> '200000000000000000000000000000000000'

This value should match the value when you run the following on the old protocol:

```ts
const [dev, property] = await init()
await mine(1)
await dev.dev.deposit(property.address, 100)
await mine(1)
await dev.dev.deposit(property.address, 200)
const value = await dev.lockup.getStorageLastStakedInterestPrice(property.address, deployer)
// value
```